File: SecondYear/semestrul4/MPP/PROIECT/CSharpProjectNetworking/Server/service/GlobalService.cs

using Persistence.database;

namespace Server;

public static class GlobalService
{
    private static MainService network ;

    public static MainService GetNetwork()
    {
        if (network == null)
        {
            var organizatorRepository = new OrganizatorRepository();
            var participantRepository = new ParticipantRepository();
            var probaRepository = new ProbaRepository();
            var inscriereRepository = new InscriereRepository();
            var categorieVarstaRepository = new CategorieVarstaRepository();

            var organizatorService = new OrganizatorService(organizatorRepository);
            var participantService = new ParticipantService(participantRepository);
            var probaService = new ProbaService(probaRepository);
            var inscriereService = new InscriereService(inscriereRepository);
            var categorieVarstaService = new CategorieVarstaService(categorieVarstaRepository);

            network = new MainService(organizatorService, participantService, probaService, inscriereService, categorieVarstaService);
        }
        return network;
    }
}

File: SecondYear/semestrul4/MPP/PROIECT/JavaProjectORM/ProjectServer/src/main/java/ro/mpp2025/javaprojectui/service/GlobalService.java
package ro.mpp2025.javaprojectui.service;

import ro.mpp2025.javaprojectui.database.*;
import ro.mpp2025.javaprojectui.database.hibernate.OrganizatorRepositoryHibernate;
import ro.mpp2025.javaprojectui.database.hibernate.ParticipantRepositoryHibernate;
import ro.mpp2025.javaprojectui.interfaces.IOrganizatorRepository;
import ro.mpp2025.javaprojectui.interfaces.IParticipantRepository;

import java.sql.SQLException;

public class GlobalService {
    private static MainService network = null;

    private GlobalService() {
    }

    public static MainService getNetwork() throws SQLException {
        if (network == null) {
//            OrganizatorRepository organizatorRepository = new OrganizatorRepository();
//            ParticipantRepository participantRepository = new ParticipantRepository();
            IOrganizatorRepository organizatorRepository = new OrganizatorRepositoryHibernate();
            IParticipantRepository participantRepository = new ParticipantRepositoryHibernate();
            ProbaRepository probaRepository = new ProbaRepository();
            InscriereRepository inscriereRepository = new InscriereRepository();
            CategorieVarstaRepository categorieVarstaRepository = new CategorieVarstaRepository();


            OrganizatorService organizatorService = new OrganizatorService(organizatorRepository);
            ParticipantService participantService = new ParticipantService(participantRepository);
            ProbaService probaService = new ProbaService(probaRepository);
            InscriereService inscriereService = new InscriereService(inscriereRepository);
            CategorieVarstaService categorieVarstaService = new CategorieVarstaService(categorieVarstaRepository);

            network = new MainService(organizatorService, participantService, probaService, inscriereService, categorieVarstaService);
        }
        return network;
    }
}
